network.go

```diff
@@ -957,10 +989,36 @@ func (n *network) delete(force bool) error {
 		return &UnknownNetworkError{name: name, id: id}
 	}
 
-	if len(n.loadBalancerIP) != 0 {
-		endpoints := n.Endpoints()
-		if force || (len(endpoints) == 1 && !n.ingress) {
-			n.deleteLoadBalancerSandbox()
+	// only remove ingress on force removal or explicit ingress removal
+	if n.ingress && !force && !delIngress {
+		return &ActiveEndpointsError{name: n.name, id: n.id}
+	}
+
+	// Check that the network is empty
+	var emptyCount uint64
+	if n.hasLoadBalancerEndpoint() {
+		emptyCount = 1
+	}
+	if !force && n.getEpCnt().EndpointCnt() > emptyCount {
+		if n.configOnly {
+			return types.ForbiddenErrorf("configuration network %q is in use", n.Name())
+		}
+		return &ActiveEndpointsError{name: n.name, id: n.id}
+	}
+
+	if n.hasLoadBalancerEndpoint() {
+		// If we got to this point, then the following must hold:
+		//  * force is true, OR
+		//  * endpoint count == 1 AND:
+		//      * n.ingress is false OR
+		//      * delIngress is true OR
+		//      * both are true
+		if err := n.deleteLoadBalancerSandbox(); err != nil {
+			if !force {
+				return err
+			}
+			// continue deletion when force is true even on error
+			logrus.Warnf("Error deleting load balancer sandbox: %v", err)
```

Comment:
I believe I was just keeping it consistent with the previous behavior which put it at the warning level.  Presumably this behavior was because failure to remove may be due to the fact that it already got inadvertantly removed.   In this case, the error is not fatal to the system since the network is being removed anyway.  (see the comment above the warning).  If this is the rationale, then warning level seems appropriate.  The error message was, I think, the same, but we could change the wording in principle.